Common/Python/PandasConverter.cs

```diff
@@ -226,5 +213,74 @@ private PyObject MakeIndicatorDataFrame(PyDict pyDict)
         {
             return _pandas.DataFrame(pyDict, columns: pyDict.Keys().Select(x => x.As<string>().ToLowerInvariant()).OrderBy(x => x));
         }
+
+        /// <summary>
+        /// Gets the <see cref="PandasData"/> for the given symbol if it exists in the dictionary, otherwise it creates a new instance with the
+        /// given base data and adds it to the dictionary
+        /// </summary>
+        private PandasData GetPandasDataValue(IDictionary<Symbol, PandasData> sliceDataDict, Symbol symbol, object data, ref int maxLevels)
+        {
+            PandasData value;
+            if (!sliceDataDict.TryGetValue(symbol, out value))
+            {
+                sliceDataDict.Add(symbol, value = new PandasData(data));
+                maxLevels = Math.Max(maxLevels, value.Levels);
+            }
+
+            return value;
+        }
+
+        /// <summary>
+        /// Adds each slice data to the pandas data dictionary
+        /// </summary>
+        private void AddSliceDataToDict(Slice slice, IDictionary<Symbol, PandasData> sliceDataDict, ref int maxLevels)
+        {
+            foreach (var key in slice.Keys)
+            {
+                var baseData = slice[key];
+                var value = GetPandasDataValue(sliceDataDict, key, baseData, ref maxLevels);
+
+                if (value.IsCustomData)
+                {
+                    value.Add(baseData);
+                }
+                else
+                {
+                    var ticks = slice.Ticks.ContainsKey(key) ? slice.Ticks[key] : null;
+                    var tradeBars = slice.Bars.ContainsKey(key) ? slice.Bars[key] : null;
+                    var quoteBars = slice.QuoteBars.ContainsKey(key) ? slice.QuoteBars[key] : null;
+                    value.Add(ticks, tradeBars, quoteBars);
+
+                }
+            }
+        }
+
+        /// <summary>
+        /// Adds each slice data corresponding to the requested data type to the pandas data dictionary
+        /// </summary>
+        private void AddSliceDataTypeDataToDict(Slice slice, Type dataType, IDictionary<Symbol, PandasData> sliceDataDict, ref int maxLevels)
+        {
+            var isTick = dataType == typeof(Tick) || dataType == typeof(OpenInterest);
+            // Access ticks directly since slice.Get(typeof(Tick)) and slice.Get(typeof(OpenInterest)) will return only the last tick
+            var sliceData = isTick ? slice.Ticks : slice.Get(dataType);
+
+            foreach (var key in sliceData.Keys)
+            {
+                var baseData = sliceData[key];
+                var value = GetPandasDataValue(sliceDataDict, key, baseData, ref maxLevels);
```

Comment:
Believe that because baseData is dynamic value is being returned as dynamic, suggest replacing `var` here by the `PandasData` type we know it is, to avoid late binding when possible, same for method above. See https://stackoverflow.com/questions/7829518/dynamic-parameter-causes-compiler-to-think-method-return-is-dynamic for example